src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignmentRegion.java

```diff
@@ -0,0 +1,187 @@
+        this.contigId = contigId;
+        this.assemblyId = assemblyId;
+        this.forwardStrandCigar = forwardStrandCigar;
+        this.forwardStrand = forwardStrand;
+        this.referenceInterval = referenceInterval;
+        this.mqual = mqual;
+        this.startInAssembledContig = startInAssembledContig;
+        this.endInAssembledContig = endInAssembledContig;
+        this.assembledContigLength = forwardStrandCigar.getReadLength();
+        this.mismatches = mismatches;
+    }
+
+    public AlignmentRegion(final GATKRead read) {
+        this.assemblyId = null;
+        this.contigId = read.getName();
+        this.forwardStrand = ! read.isReverseStrand();
+        this.forwardStrandCigar = forwardStrand ? read.getCigar() : CigarUtils.invertCigar(read.getCigar());
+        this.referenceInterval = new SimpleInterval(read);
+        this.assembledContigLength = forwardStrandCigar.getReadLength() + getTotalHardClipping(forwardStrandCigar);
+        this.startInAssembledContig = startOfAlignmentInContig(forwardStrandCigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, forwardStrandCigar);
+        this.mqual = read.getMappingQuality();
+        if (read.hasAttribute("NM")) {
+            this.mismatches = read.getAttributeAsInteger("NM");
+        } else {
+            this.mismatches = 0;
+        }
+    }
+
+    public int overlapOnContig(final AlignmentRegion other) {
+        return Math.max(0, Math.min(endInAssembledContig + 1, other.endInAssembledContig + 1) - Math.max(startInAssembledContig, other.startInAssembledContig));
+    }
+
+    private static int getTotalHardClipping(final Cigar cigar) {
+        final List<CigarElement> cigarElements = cigar.getCigarElements();
+        return (cigarElements.get(0).getOperator() == CigarOperator.HARD_CLIP ? cigarElements.get(0).getLength() : 0) +
+                (cigarElements.get(cigarElements.size() - 1).getOperator() == CigarOperator.HARD_CLIP ? cigarElements.get(cigarElements.size() - 1).getLength() : 0);
+    }
+
+    private static int startOfAlignmentInContig(final Cigar cigar) {
+        return getClippedBases(true, cigar) + 1;
+    }
+
+    private static int endOfAlignmentInContig(final int assembledContigLength, final Cigar cigar) {
+        return assembledContigLength - getClippedBases(false, cigar);
+    }
+
+    private static int getClippedBases(final boolean fromStart, final Cigar cigar) {
+        int posInContig = 0;
+        int j = fromStart ? 0 : cigar.getCigarElements().size() - 1;
+        final int offset = fromStart ? 1 : -1;
+        CigarElement ce = cigar.getCigarElement(j);
+        while (ce.getOperator().isClipping()) {
+            posInContig += ce.getLength();
+            j += offset;
```

Comment:
The cigar is arguably broken if it contains only clipping, but still, you might add:
if ( j < 0 || j >= cigar.getCigarElements().size() ) break;